__init__.py

```diff
@@ -354,10 +354,16 @@ def _describe_routine(self, message):
 
     @intent_handler(IntentBuilder("ScheduleRoutine").require("Schedule").require("RoutineName"))
     def _add_routine_schedule(self, message):
-        name = message.data["RoutineName"]
-        days = self._get_days()
-        hour, minute = self._get_time()
-        cronstring = self._generate_cronstring(days, hour, minute)
+        name = message.data["RoutineName"] 
+        try:
+            days = self._get_days()
+            hour, minute = self._get_time()
+            if days == None or len(days)==0: 
+                self.speak_dialog("unknown.day")
+                return
+            cronstring = self._generate_cronstring(days, hour, minute)
+        except:
```

Comment:
Ok, I've looked more carefully.  First, thanks for the bit about except Exception.  Python is relatively new to me, so anything I can learn is good.  

But second, if there was an exception in either get_days or get_time why would you want to update the schedule and write out the file?  Something is wrong!  Note that a bad time is already caught, and the error spoken in _get_time, at which time it returns None and causes an exception at hour, minute = self._get_time since it is not a list.  Nothing caught the exception, and we got an ugly spoken message about an exception in the skill.   Similarly, if _get_days did not find a valid day, it returned either None or something with length 0.  In this case, I believe cronstring threw an error.

I think that at least one of the corruption cases came if one did not speak the time fast enough, but now I'm not sure.  It may be that explicitly catching those errors consistently in the "_get_xxx" routines would be best, but then we'd want to either throw and catch a specific error from them (and return when it was caught) or test the return value for None or len=0.  I'd think a try/except would be simpler, but I would probably throw and catch only a value error or something like that and only catch it around the two _get routines.

I'm taking a break now, but I'll a) remove the exception stuff and reproduce the file corruption and b) if I don't hear from you in an hour or 2, implement that error catching as I suggested above (assuming it catches the corruption and exceptions).  If you can't respond that soon and don't like what I did, that's ok.  I'll re-do it.  It's your skill!

Reply:
> if there was an exception in either get_days or get_time why would you want to update the schedule and write out the file?

We don't/can't. Because it doesn't currently catch the exception, it can never get to the lines where it writes the file. You could show this by editing one of those `_get_X` methods to always throw an exception, and adding a log line that logs something after `_get_X` has been called, but before the file is written. You'll see that the exception results in nothing after it being executed, so you won't see your log message.

If you do the try/except, and just return, yes, you will avoid mycroft speaking the message about an error occurring, however, it doesn't change whether the file is written or not. In both cases (with/without the try/except), those lines will not be executed if there is an error.

If the goal is just to prevent the "ugly" message from mycroft on error, that's fine, but we should do something other than just `return`. E.g. we should log the error, and say something like "I didn't understand that, please try again." Otherwise there is not indication to the user that there was a problem (assuming it's not caught elsewhere).